src/main/fc/config.c

```diff
@@ -1047,17 +1047,26 @@ void validateAndFixGyroConfig(void)
     float samplingTime = 0.000125f;
 
     if (gyroConfig()->gyro_use_32khz) {
-#ifdef GYRO_SUPPORTS_32KHZ
-        samplingTime = 0.00003125;
         // F1 and F3 can't handle high pid speed.
 #if defined(STM32F1)
         pidConfig()->pid_process_denom = constrain(pidConfig()->pid_process_denom, 16, 16);
-#endif
-#if defined(STM32F3)
+        gyroConfig()->gyro_use_32khz = false;
+#elif defined(STM32F3)
         pidConfig()->pid_process_denom = constrain(pidConfig()->pid_process_denom, 4, 16);
-#endif
-#else
         gyroConfig()->gyro_use_32khz = false;
+#else
+        switch (detectedSensors[SENSOR_INDEX_GYRO]) {
+        default:
+            gyroConfig()->gyro_use_32khz = false;
+            break;
+        case GYRO_MPU6500:
+        case GYRO_MPU9250:
+        case GYRO_ICM20689:
+        case GYRO_ICM20608G:
+        case GYRO_ICM20602:
+            samplingTime = 0.00003125;
+            break;
+        }
 #endif
```

Comment:
The above switch will need to be moved to here: https://github.com/betaflight/betaflight/blob/master/src/main/drivers/accgyro_mpu.c#L172 
perhaps